src/Microsoft.Identity.Client/Instance/Discovery/KnownMetadataProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Instance.Discovery
+{
+    internal class KnownMetadataProvider : IKnownMetadataProvider
+    {
+        // No need to use a ConcurrentDictionary, because the normal Dictionary is thread safe for read operations
+        private static readonly IDictionary<string, InstanceDiscoveryMetadataEntry> s_knownEntries =
+            new Dictionary<string, InstanceDiscoveryMetadataEntry>();
+
+        private static readonly ISet<string> s_knownEnvironments = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        static KnownMetadataProvider()
+        {
+            InstanceDiscoveryMetadataEntry publicCloudEntry = new InstanceDiscoveryMetadataEntry()
+            {
+                Aliases = new[] { "login.microsoftonline.com", "login.windows.net", "login.microsoft.com", "sts.windows.net" },
+                PreferredNetwork = "login.microsoftonline.com",
+                PreferredCache = "sts.windows.net"
+            };
+
+            InstanceDiscoveryMetadataEntry cnCloudEntry = new InstanceDiscoveryMetadataEntry()
```

Comment:
>cn [](start = 43, length = 2)

i would write these out so it's clear to someone looking at this who these env are for

Reply:
I don't understand this comment

Reply:
like put `chinaCloudEntry` instead of `cnCloudEntry`...just a suggestion

Reply:
Ok, makes sense.